safekeeper/src/wal_service.rs

```diff
@@ -5,25 +5,32 @@
 use anyhow::Result;
 use regex::Regex;
 use std::net::{TcpListener, TcpStream};
+use std::sync::Arc;
 use std::thread;
 use tracing::*;
+use utils::auth::JwtAuth;
 
 use crate::handler::SafekeeperPostgresHandler;
 use crate::SafeKeeperConf;
 use utils::postgres_backend::{AuthType, PostgresBackend};
 
 /// Accept incoming TCP connections and spawn them into a background thread.
-pub fn thread_main(conf: SafeKeeperConf, listener: TcpListener) -> Result<()> {
+pub fn thread_main(
+    conf: SafeKeeperConf,
+    listener: TcpListener,
+    auth: Option<Arc<JwtAuth>>,
```

Comment:
Let's move public key inside `SafekeeperConf`.

Reply:
Will do together with corresponding Pageserver refactoring in a separate PR.